server/src/com/cloud/tags/TaggedResourceManagerImpl.java

```diff
@@ -215,6 +215,10 @@ public long getResourceId(String resourceId, ResourceObjectType resourceType) {
             }
         }
 
+        if (resourceType == ResourceObjectType.Project) {
+            accountId = ((ProjectVO)entity).getProjectAccountId();
```

Comment:
Couple of questions -
1) Is it safe to assume the project account id will map to the user who created the project? Looks like it is set up from the passed parameter to Create Project Command. 
2) What happens to the projects created directly via command line. Do they work as expected? (Asking this as it is missing in the tests conducted in the PR Description.)

Also, the bug of users not being able to tag their resources extends to some other types of resources (as far as I recall in the mailing lists) but I guess we should file an open issue for others as it would be outside the scope of the current issue.

Reply:
Tested using cloudmonkey 

(local) mycloudmonkey> create tags resourcetype=project resourceids=5c3eced9-1b05-494a-8f65-72565f491d27 tags[0].key="can't stop" tags[0].value="time is money"

accountid = d16af203-c08d-4157-bf5a-94be8f302390
cmd = org.apache.cloudstack.api.command.user.tag.CreateTagsCmd
created = 2019-05-09T15:30:19+0000
jobid = de83c6b2-a9e7-4b3f-b8b5-49c55cc81377
jobprocstatus = 0
jobresult:
success = True
jobresultcode = 0
jobresulttype = object
jobstatus = 1
userid = 6d1bcfeb-0640-487e-978a-f4cf348accb9